### PR TITLE
Remove INamespacePartitioningStrategy, ICompositionStrategy, ISanitizationStrategy and IHandleOversizedBrokeredMessages from the container

### PR DESCRIPTION
**Pull request by danielmarbach**

### CODE REVIEW

File: src/Tests/API/APIApprovals.ApproveAzureServiceBusTransport.approved.txt [hunk 1]
```diff
@@ -65,8 +65,7 @@ namespace NServiceBus
         public NServiceBus.AzureServiceBusMessageSenderSettings BackOffTimeOnThrottle(System.TimeSpan backoffTime) { }
         public NServiceBus.AzureServiceBusMessageSenderSettings MaximuMessageSizeInKilobytes(int sizeInKilobytes) { }
         public NServiceBus.AzureServiceBusMessageSenderSettings MessageSizePaddingPercentage(int percentage) { }
-        public NServiceBus.AzureServiceBusMessageSenderSettings OversizedBrokeredMessageHandler<T>(T instance)
```
Comment (danielmarbach) [inline src/Tests/API/APIApprovals.ApproveAzureServiceBusTransport.approved.txt]: Discussed this with @yvesgoelevan and we figured it is OK to do this in a major. There is no way we can deprecate this API properly that's why it was directly replaced

**Comment by SeanFeldman** [comment]
It fails on flaky tests (deferral), not the changes in the PR.

File: src/Transport/Topology/Topologies/TypeExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+{
+    using System;
+    using System.Linq;
+    using System.Reflection;
+    using Settings;
+
+    static class TypeExtensions
+    {
+        // potentially we want to remove this when we simplify hooking customer owned implementations into the topology
+        public static TReturnedType CreateInstance<TReturnedType>(this Type typeToResolve, ReadOnlySettings settings)
```
Comment (danielmarbach) [inline src/Transport/Topology/Topologies/TypeExtensions.cs]: This keeps backwards compatibility:

Following assumptions. Right now with the transport parts container it is possible to:

- Resolve dependencies that are in the parts container (all of them are internal)
- Resolve ReadonlySettings
- Have empty default ctor

From a customer perspective it is only possible to do the following:

- have a ctor on their implementations which required ReadonlySettings or
- have an empty parameterless ctor

Therefore this extension methods keeps this by using either default ctor and calling or assume ctor that has a dependency to readonly settings. In all other cases it blows up

**Comment by danielmarbach** [comment]
Rebased

**Comment by SeanFeldman** [comment]
@yvesgoeleven is marked as a reviewer as well. Randomly failing non-related tests are **not** affected by the changes in this PR.

**Comment by danielmarbach** [comment]
@seanfeldman yves and I did this together. Good to go

**Comment by SeanFeldman** [comment]
Merged.

**Pull request merged**